test/statsd/ecs_statsd_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package statsd
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+)
+
+var _ test_runner.IECSTestRunner = (*ECSStatsdTestRunner)(nil)
+
+const testRetryCount = 3
+
+type ECSStatsdTestRunner struct {
+	test_runner.ECSBaseTestRunner
+}
+
+func (t *ECSStatsdTestRunner) Validate(e *environment.MetaData) status.TestGroupResult {
+	metricsToFetch := t.GetMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, metricName := range metricsToFetch {
+		result := t.validateStatsdMetric(metricName)
+
+		// there seems to be some delay before the runner is able to fetch metrics from CW
+		if result.Status != status.SUCCESSFUL {
+			for j := 0; j < testRetryCount; j++ {
+				time.Sleep(15 * time.Second)
+				result = t.validateStatsdMetric(metricName)
+				if result.Status == status.SUCCESSFUL {
+					break
+				}
+			}
+		}
```

Comment:
we should just remove the first validation check and just do the retry.

Reply:
yup, agree